starter_code/elevator.js

```diff
@@ -1,19 +1,99 @@
 class Elevator {
-  constructor(){
-    this.floor      = 0;
-    this.MAXFLOOR   = 10;
-    this.requests   = [];
-  }
-
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  constructor() {
+    this.floor = 0;
+    this.MAXFLOOR = 10;
+    this.waitingList = [];
+    this.passengers = [];
+    this.requests = [];
+    this.direction = "";
+  }
+
+  start() {
+    this.startInterval = setInterval(() => {
+      this.update();
+    }, 1000);
+  }
+
+  stop() {
+    clearInterval(this.startInterval);
+  }
+
+  update() {
+    for (let i = 0; i < this.MAXFLOOR; i++) {
```

Comment:
As we discussed yesterday, I would regard the `update` method as a function that is invoked at every floor, since it's being called at every interval. So, I wouldn't loop over all the floors, just check the state of the current floor, and call the methods accordingly.

Reply:
But this also depends on how you actually use all these methods in the `index.js` file. Anyways, it's fine as long as you exercised some asynchronous concepts.